platform/src/main/java/org/hillview/utils/IntSet.java

```diff
@@ -94,20 +98,21 @@ public boolean add(final int k) {
         if (this.size++ >= this.maxFill) {
             this.rehash(HashUtil.arraySize(this.size + 1, this.f));
         }
+        this.iteratorKey = this.key;
         return true;
     }
 
     /**
      *
      * @param pos a location in the key[] array
-     * @return the location of the next full slot after cursor.
+     * @return the location of the next full slot after cursor. Operates on the iteratorKey array
```

Comment:
In general you should not expose in such a comment information about a private field.
The comment for an API function should be read as describing what the function does, but not how it does it - unless it's really important for the caller to know this.

Reply:
It is important, because if the iterator is sorted this gives a sorted view which does not correspond to the hashing. This function exposes an internal array, so one may argue that in a perfect design this function is not implemented at all, but in our case it seems like a reasonable tradeoff for the optimization we do.